File: src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3

# Line bellow is for making this file an executable

import rclpy
from rclpy.node import Node

# Using POO
class MyNode(Node):
    
    def __init__(self):
        super().__init__("py_test")
        self.counter_ = 0
        self.get_logger().info("Hello, I'm working with POO on ROS2!!!")
        self.create_timer(0.5, self.timer_callback) # create_timer(seconds, function_to_callback)

    # A callback function
    def timer_callback(self):
        self.counter_ += 1
        self.get_logger().info("Hello!! " + str(self.counter_))

def main(args=None):
    # Initilize the ROS2 communication
    rclpy.init(args=args)

    # Create an object of Node("name_of_node")
    node = MyNode()
    #node = Node("py_test")
    #node.get_logger().info("Hello, I'm working on ROS2")
    
    # rclpy.spin allows to mantain the programa alive
    rclpy.spin(node)
    
    # Shutdown the ROS2 communication
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

// Using POO

class MyNode: public rclcpp::Node
{
public:
    MyNode(): Node("cpp_test"), counter_(0){
        RCLCPP_INFO(this->get_logger(), "Hello from ROS2 using C++");

        // Calling the timer function
        timer_ = this->create_wall_timer(std::chrono::seconds(1), 
                                         std::bind(&MyNode::timeCallback, this));
    }

private:

    void timeCallback(){
        counter_ ++;
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_);
    }

    // Function fo the timer with SharedPointer 
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};

int main(int argc, char **argv){

    // Initialize the ROS2 communication
    rclcpp::init(argc, argv);

    // Using shared pointer to create the node
    auto node = std::make_shared<MyNode>();
    //auto node = std::make_shared<rclcpp::Node>("cpp_test");
    //RCLCPP_INFO(node->get_logger(), "Hello from ROS2 using C++");

    // Using spin function
    rclcpp::spin(node);

    // Shutdown ROS2 communication
    rclcpp::shutdown();

    return 0;
}